Example app

pjanik opened the pull request:
[#172225396]

I'll extend it later to support readWriteTokens and other approaches. The code is intentionally minimal, so there's no error handling, no progress spinners :wink: and other things that will just add more lines.
Client lib is pointing `..` in the `package.json`. I thought it's a bit simpler than forcing dev to use npm link instead.

TravisCI isn't set up yet, but this is how it looks:
<img width="877" alt="Screenshot 2020-04-09 at 14 10 23" src="https://user-images.githubusercontent.com/767857/78893600-e75de080-7a6b-11ea-842c-a971239f6a28.png">

scytacki commented on example-app/src/index.tsx:
```diff
@@ -0,0 +1,104 @@
+        <h3>Portal Setup</h3>
+        <p className="hint">
+          Token Service requires FirebaseJWT that can be obtained from Portal. Provide Portal configuration, click
+          "Authorize in Portal" (note that it'll reload this page) and finally "Get FirebaseJWT". When FirebaseJWT
```
You might update this description to explain that it will require a login at the portal if your user is not currently logged in.
I'd also suggest changing "finally" to "and then click".

pjanik commented:
Thanks. I've addressed the comments except for `useLocalStorage`. I tried to use this library, but it's a mess actually. It doesn't work for strings because of the bug (or intentionally). I was debugging that for an hour and gave up. The latest published version is 2.3.0 and it has this line:
https://github.com/rehooks/local-storage/blob/v2.2.2/src/use-localstorage.ts#L74
(tag is 2.2.2 as they don't have 2.3.0 tag...)
`tryParse(storedValue) !== storedValue` won't work as they expected for string value. Same strings will return true for === check.
It seems it's been changed in master, but still won't work for strings, as random string isn't a valid JSON:
https://github.com/rehooks/local-storage/blob/master/src/use-localstorage.ts#L88
In the end `canWrite` is true and localStorage value is overwritten by initialValue. 
Another problem is typings. Even when the initial value is provided, the types indicate that this hook can return `null` which enforces crazy checks everywhere in the code or manual casting. 
So, I gave up. ;) I'll keep it in mind and if they release a new version or if I see something similar, I'll try to use it. I've seen https://www.npmjs.com/package/react-use-localstorage too, but they don't have `writeStorage` so the lines that you mentioned would need to stay.
